handlers/pipeline.go

```diff
@@ -136,6 +136,50 @@ func PipelineGet(c echo.Context) error {
 	return c.String(http.StatusNotFound, errPipelineNotFound.Error())
 }
 
+// PipelineDelete accepts a pipeline id and deletes it from the
+// store. It also removes the binary inside the pipeline folder.
+func PipelineDelete(c echo.Context) error {
+	pipelineIDStr := c.Param("pipelineid")
+
+	pipelineID, err := strconv.Atoi(pipelineIDStr)
+	if err != nil {
+		return c.String(http.StatusBadRequest, errInvalidPipelineID.Error())
+	}
+
+	// Look up pipeline for the given id
+	var foundPipeline *gaia.Pipeline
+	var index int
+	var deletedPipelineIndex int
+	for pipeline := range pipeline.GlobalActivePipelines.Iter() {
+		if pipeline.ID == pipelineID {
+			foundPipeline = &pipeline
+			deletedPipelineIndex = index
+		}
+		index++
+	}
+
+	if foundPipeline == nil {
+		return c.String(http.StatusNotFound, err.Error())
+	}
+
+	// Delete pipeline binary
+	err = pipeline.DeleteBinary(*foundPipeline)
+	if err != nil {
+		return c.String(http.StatusInternalServerError, err.Error())
```

Comment:
Instead of an internal server error I suggest creating an `errPipelineDelete` error.  Something like, `oops sorry, I couldn't delete the binary, maybe permissions are incorrect? Please fix permissions and try again or delete it by hand.`. Thoughts?